qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -90,6 +112,108 @@ def quantum_instance(
         else:
             self._quantum_instance = quantum_instance
 
+    @property
+    def user_parameters(self) -> Union[ParameterVector, Sequence[Parameter]]:
+        """Return the vector of user parameters."""
+        return self._user_parameters
+
+    @user_parameters.setter
+    def user_parameters(self, user_params: Union[ParameterVector, Sequence[Parameter]]) -> None:
+        """Sets the user parameters"""
+        if user_params is None:
+            self._user_param_binds = None
+        else:
+            self._user_param_binds = {
+                user_params[i]: user_params[i] for i, _ in enumerate(user_params)
+            }
+
+        self._user_parameters = user_params
+
+    def assign_user_parameters(
+        self, values: Union[Mapping[Parameter, ParameterValueType], Sequence[ParameterValueType]]
+    ) -> None:
+        """
+        Assign user parameters in the QuantumKernel feature map.
+
+        Args:
+            values (dict or iterable):
+                Dict {parameter: value, ...} or {parameter: parameter}:
+                The keys of the parameter dictionary must be Parameter instances in the
+                feature map circuit. The values of the dictionary should be either numeric
+                values {param:float} or copies of the param key {param:param}. Passing a
+                copy of the param key as a value causes that parameter to be ignored.
+                This allows the circuit's user parameters to be bound consecutively.
+
+                Iterable [value_0, value_1, ..., value_N]:
+                If a list of real values is passed, the elements are assigned to the parameters
+                stored in the user_parameters field.
+
+        Return:
+            None
+
+        Raises:
+            ValueError: Incompatible number of user parameters and values
+        """
+        if not isinstance(values, dict):
+            if self._user_parameters is None:
+                raise ValueError(
+                    """
+                    An iterable of user parameter values was given, but no user
+                    parameters were defined for the QuantumKernel.
+                    """
+                )
```

Comment:
Should this check be outside of the above check of whatever the type of values is? No matter the type, if there were no user defined parameters set then this should raise an error right independent of the form of values being passed in?

Reply:
If an iterable is passed, but no user parameters have been specified, we throw an error because we have no information for how to map the values.

If a dict is passed, we are able to find the values to the Parameters "blindly" by simply passing them straight down to the QuantumCircuit API for assignment/binding.

Reply:
If there are no user parameters on the kernel though why would we let it bind that way - seems odd to let the user bind parameters when the kernel has none even though you might not need an additional code as such for the binding. Arguably this way you can bind parameters without ever defining them as user parameters - or am I missing something